code/datums/abilities/flock/flockmind.dm

```diff
@@ -459,12 +459,11 @@
 	if(..())
 		return TRUE
 	var/mob/living/intangible/flock/F = holder.owner
-	//special handling for building ghosts
-	if(istype(target,/obj/flock_structure/ghost))
-		//do the tgui window instead
-		//this actually doesn't need bonus behaviour because the cancelbuild is on click, but will need to fix this if we change that in future
-		return TRUE
-	else if(HAS_ATOM_PROPERTY(target,PROP_ATOM_FLOCK_THING)) //it's a thing we've converted, we can deconstruct it
+	if(HAS_ATOM_PROPERTY(target,PROP_ATOM_FLOCK_THING))
+		if(istype(target,/obj/flock_structure/ghost))
+			return TRUE // does tgui window for canceling since cancelbuild is on click, will need to fix this if we change it in the future
+		if (istype(target, /mob/living/critter/flock) || istype(target, /turf/simulated/floor/feather) || istype(target, /obj/flock_structure/rift) || istype(target, /obj/flock_structure/egg) || istype(target, /obj/flock_structure/relay))
```

Comment:
```suggestion
		if (ismob(target) || istype(target, /turf/simulated/floor/feather) || istype(target, /obj/flock_structure/rift) || istype(target, /obj/flock_structure/egg) || istype(target, /obj/flock_structure/relay))
```
let's just assume we don't want to deconstruct any mobs

Reply:
I think the atom property should handle that, but can still add if wanted.

Reply:
I'd prefer some sort of generic way to tag deconstruction blacklist, instead of this copy+pasted if

Reply:
Done, though mob attacking is handled by harm() instead of attack_hand(), even though checked for there in the macro